.github/workflows/test.yml

```diff
@@ -46,3 +49,22 @@ jobs:
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
         uses: codecov/codecov-action@v3
+
+  # Only run deploy on push to main when tests succeed
```

Comment:
How about: "Run deploy on push to main only if tests succeed"?

Reply:
That's certainly a better way to put it